### PR TITLE
Consolidate cleanup for Azure blob tests

### PR DESCRIPTION
**Pull request by reta**
### Description
Consolidate cleanup for Azure blob tests

### Related Issues
Followup on https://github.com/opensearch-project/OpenSearch/pull/16778

### Check List
- [ ] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugins/repository-azure/src/test/java/org/opensearch/repositories/azure/AzureRepositorySettingsTests.java [hunk 1]
```diff
@@ -57,6 +58,7 @@
 public class AzureRepositorySettingsTests extends OpenSearchTestCase {
     @AfterClass
     public static void shutdownSchedulers() {
+        HttpResources.disposeLoopsAndConnections();
```
Comment (reta) [inline plugins/repository-azure/src/test/java/org/opensearch/repositories/azure/AzureRepositorySettingsTests.java:61]: @cwperks I am not sure we have to block here

Comment (cwperks) [reply plugins/repository-azure/src/test/java/org/opensearch/repositories/azure/AzureRepositorySettingsTests.java:61]: Thank you for fixing this @reta! I should have switched the PR to draft as I was unsure of this cleanup block. I saw that it fixed the thread leak, but didn't quite understand why.

I saw that it was stuck on NIOEventLoopGroup. Does this call remove all EventLoops in the group?

Comment (reta) [reply plugins/repository-azure/src/test/java/org/opensearch/repositories/azure/AzureRepositorySettingsTests.java:61]: > I saw that it was stuck on NIOEventLoopGroup. Does this call remove all EventLoops in the group?

It should, let's see if tests pass

**Comment by reta** [comment]
@cwperks seems like we are good, @andrross @dblock mind please quick look? thanks!

**Review by cwperks — COMMENTED** [review]
Approved. Leaving a comment instead of marking as approved since I'm not a maintainer, but in the admin group.

**Comment by reta** [comment]
@dblock @andrross could you folks help me to get this one in? thank you

**Review by andrross — APPROVED** [review]

**Pull request merged**